src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -667,6 +668,62 @@ def create_application(client, display_name, homepage=None, identifier_uris=None
     return result
 
 
+def grant_application(cmd, identifier, appid, expires='1'):
+    if not appid:
+        raise CLIError("'--appid' is required for grant OAuth2 permissions")
+
+    graph_client = _graph_client_factory(cmd.cli_ctx)
+
+    # Get the Service principal ObjectId for the client app
+
+    sps = list(graph_client.service_principals.list(
+        filter="appId eq '{}'".format(identifier)))
```

Comment:
I would suggest you just call `list_sps`  in the same file.

Reply:
Extended ```list_sps``` to include search by appid/clientid

```
def list_sps(client, spn=None, display_name=None, query_filter=None, appid=None):
    sub_filters = []
    if query_filter:
        sub_filters.append(query_filter)
    if spn:
        sub_filters.append("servicePrincipalNames/any(c:c eq '{}')".format(spn))
    if display_name:
        sub_filters.append("startswith(displayName,'{}')".format(display_name))
    if appid:
        sub_filters.append("appId eq '{}'".format(appid))

    return client.list(filter=(' and '.join(sub_filters)))
```